modules/layers/src/text-layer/utils.js

```diff
@@ -69,42 +73,110 @@ export function buildMapping({
   };
 }
 
-export function transformRow(row, iconMapping, lineHeight) {
+export function autoWrapping({
+  string,
+  lineHeight,
+  width,
+  wordBreak,
+  fontSize,
+  fontFamily,
+  fontWeight,
+  textAlign
+}) {
+  const characters = Array.from(string);
+  const textDiv = document.createElement('div');
+  textDiv.id = AUTO_WRAPPING_DIV_ID;
+  textDiv.style = `
+    ${AUTO_WRAPPING_DIV_STYLE}
+    word-break: ${wordBreak};
+    lineHeight: ${lineHeight};
+    width: ${width}px;
+    font-size: ${fontSize}px;
+    font-family: ${fontFamily};
+    font-weight: ${fontWeight};
+    text-align: ${textAlign || 'left'}
+  `;
+
+  characters.map(character => {
```

Comment:
Need some perf test on which is faster - `appendChild` or just setting `innerHTML`.

Reply:
innerHTML works way better